### ISSUE TITLE
GCC + INET 3.6.1: PacketDrillApp.cc compile error

### ISSUE THREAD

**Issue by sxleixer** [inet-framework/inet#265]
I'm compiling INET 3.6.1 with gcc.

This yields the following error:
```
inet/applications/packetdrill/PacketDrillApp.cc:969:45: error: ordered comparison between pointer and zero ('PacketDrillExpression *' and 'int')
            if (expr_params->spp_hbinterval > 0)
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~
inet/applications/packetdrill/PacketDrillApp.cc:971:45: error: ordered comparison between pointer and zero ('PacketDrillExpression *' and 'int')
            if (expr_params->spp_pathmaxrxt > 0)
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~
```

I fixed the issue with the following patch:
```
--- src/inet/applications/packetdrill/PacketDrillApp.cc	2017-08-25 16:37:50.581467727 +0200
+++ src/inet/applications/packetdrill/PacketDrillApp.new.cc	2017-08-25 16:39:59.087674141 +0200
@@ -966,9 +966,9 @@
                 sctpSocket.setEnableHeartbeats(false);
             else if (expr_params->spp_flags->getNum() & SPP_HB_ENABLE)
                 sctpSocket.setEnableHeartbeats(true);
-            if (expr_params->spp_hbinterval > 0)
+            if (expr_params->spp_hbinterval > (void *) 0)
                 sctpSocket.setHbInterval(expr_params->spp_hbinterval->getNum());
-            if (expr_params->spp_pathmaxrxt > (void *) 0)
+            if (expr_params->spp_pathmaxrxt)
                 sctpSocket.setPathMaxRetrans(expr_params->spp_pathmaxrxt->getNum());
             break;
         }
```

Another approach would be `if (expr_params->spp_pathmaxrxt)` which also should be a feasible solution.

**Comment by rhornig** [comment]
This is already fixed on master.

**Comment by sxleixer** [comment]
At least in the github master branch it isn't. I just checked it right now.

**Comment by rhornig** [comment]
Ups, you are right. I remembered incorrectly. It was committed on the integration branch: 147b3762793744d6c803bb334ad90bad026f4e3a

We may need to cherrypick this comment back to the master. Until them use the fix provided in this commit.

**Comment by rhornig** [comment]
Fixed both in 3.6.2 and 4.x
